### PR TITLE
deluge, ffsync, rutorrent and squidguard build fixes

### PR DESCRIPTION
**Pull request by th0ma7**
_Motivation:_  Effort in order to remove build failures from master related to `squidguard` and friends.  While at it adding multiple updates to dependencies. This PR is not intented to make the package to actually work on neither DSM6/7 but solely focus on removing build errors.
_Linked issues:_  github action from #4658

### Checklist
- [x] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [ ] New installation of package completed successfully

### ffsync
This project is no longer maintained.
This is now superseeded with a rust rewrite: https://github.com/mozilla-services/syncserver/issues/165#issuecomment-584571947
Project page: https://github.com/mozilla-services/syncstorage-rs/

### CODE REVIEW

**Comment by th0ma7** [comment]
@hgy59 feel free to chim-in.  Doing to rough job as first pass.... slowly moving along.

**Comment by th0ma7** [comment]
@hgy59 your help would be appreciated as there is one thing I do not get?
I have some code in `cross/squid` for unsupported archs but auto-builds still fails on them.
I re-used that same code into `spk/squidguard` but results stays the same.
Hints on solving this would be appreciated.

File: spk/squidguard/Makefile [hunk 1]
```diff
@@ -17,6 +17,12 @@ DISPLAY_NAME = SquidGuard
 HOMEPAGE = http://www.squidguard.org/
 LICENSE  = GPL - Copyright © 1990,2007 Oracle.  All rights reserved for berkeley db library. See LICENSE file.
 
+include ../../mk/spksrc.common.mk
+
```
Comment (hgy59) [inline spk/squidguard/Makefile:21]: obsolete include.
we don't need to include this (or spksrc.arch.mk) to use arch variables with `UNSUPPORTED_ARCHS` as make resolved those variable lately.
_we need those includes for conditional checks within the Makefile only_

```suggestion
```

**Comment by th0ma7** [comment]
It seems I have faced a blocker with this PR...
Updating `c-ares` leads to a need to update `gevent` which in turn was really really late in terms of used version.
And `gevent` fails and can't find a solution yet...

**Comment by Safihre** [comment]
Just noticed this one because of #4748, since `squiguard` has been in `BETA = 1` since _2012_ 🙃, maybe this package can just be removed?

**Comment by th0ma7** [comment]
> Just noticed this one because of #4748, since `squiguard` has been in `BETA = 1` since _2012_ upside_down_face, maybe this package can just be removed?

Good point @Safihre ... but I might have solved the latest build issues, fingers crossed :)

**Comment by th0ma7** [comment]
### ffsync
This project is no longer maintained.
This is now superseeded with a rust rewrite: https://github.com/mozilla-services/syncserver/issues/165#issuecomment-584571947
Project page: https://github.com/mozilla-services/syncstorage-rs/

@hgy59 and @publicarray and others, I'm suggesting to retire the package until someone is willing to create a new one using the rust version.

**Comment by publicarray** [comment]
I'm Ok with that I did a [quick search]([search](https://github.com/SynoCommunity/spksrc/search?o=desc&p=2&q=ffsync&s=&type=issues)) and there are a few of open issues: #2985 #2748 #2093 #2816 #3124 #2747 #2828 #2873 wiki: https://github.com/SynoCommunity/spksrc/wiki/Firefox-Sync-Server-1.5

Before removing would you mind taking a look at #3849 first (might fix build issues or might not be worth the effort :man_shrugging: )?

**Comment by th0ma7** [comment]
@publicarray  Wow, I should have noticed this first.  My only intention with this PR is to remove build errors to get cleaner output from github-action.  I'll see if can steal a few things from there as a start.

**Comment by publicarray** [comment]
Haha yea most of the time Google (ddg or whatever) is a better Github search engine. Not this time though, a Github search that actually resurns something usefull :rofl:

**Comment by th0ma7** [comment]
Woot! @publicarray finally got it to build! (not necessarily to work which would require additional changes from PR #3849)

**Comment by Safihre** [comment]
@th0ma7 don't forget to merge in the master branch, for me that broke my builds...

**Comment by th0ma7** [comment]
> don't forget to merge in the master branch, for me that broke my builds...

I'm sorry @Safihre I don't follow :man_shrugging: ?  Indeed when all build errors are gone I'll ask for review and merge to master.

**Comment by Safihre** [comment]
> > don't forget to merge in the master branch, for me that broke my builds...
> 
> I'm sorry @Safihre I don't follow 🤷‍♂️ ? Indeed when all build errors are gone I'll ask for review and merge to master.

I mean to rebase your changes in this PR/branch on the current master branch, since this branch was split a long time ago.

**Comment by th0ma7** [comment]
> I mean to rebase your changes in this PR/branch on the current master branch, since this branch was split a long time ago.

Gotcha!  Yeah, I did a rebase a few days ago but thnx for the reminder.

**Comment by th0ma7** [comment]
@hgy59 and @publicarray finally got it to build successfully!!! woot!  Ready for peer-review in hope to reduce the amount of false-positive on github-action.

File: cross/squid/Makefile [hunk 2]
```diff
@@ -1,28 +1,45 @@
 PKG_NAME = squid
-PKG_VERS = 3.1.22
+PKG_VERS = 5.0.4
 PKG_EXT = tar.gz
-PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
-PKG_DIST_SITE = http://www.squid-cache.org/Versions/v3/3.1/
-PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+PKG_DIST_NAME = $(shell echo '$(PKG_NAME)' | tr '[:lower:]' '[:upper:]')_$(subst .,_,$(PKG_VERS)).$(PKG_EXT)
+PKG_DIST_SITE = https://github.com/squid-cache/squid/archive
+PKG_DIR = $(PKG_NAME)-$(shell echo '$(PKG_NAME)' | tr '[:lower:]' '[:upper:]')_$(subst .,_,$(PKG_VERS))
```
Comment (publicarray) [inline cross/squid/Makefile:6]: I'm not sure shelling out is worth it. Just using `SQUID` is good enough IMHO (same with PKG_DIST_NAME)
```suggestion
PKG_DIR = $(PKG_NAME)-SQUID_$(subst .,_,$(PKG_VERS))
```

**Review by publicarray — COMMENTED** [review]
It's good to see the green check marks :)

FYI `01-squidguardmgr.cgi.patch-DISABLE ` was renamed and is now disabled.

File: cross/squid/Makefile (hunk 2 above)
Comment (th0ma7) [reply cross/squid/Makefile:6]: I'm actually surprised I did that... let me rework that.

Comment (th0ma7) [reply cross/squid/Makefile]: solved.

**Comment by th0ma7** [comment]
> FYI `01-squidguardmgr.cgi.patch-DISABLE ` was renamed and is now disabled.

Remove as no longer needed with the newer version

**Comment by th0ma7** [comment]
@hgy59, @ymartin59  and @publicarray I've pretty much solved all I could find to make the builds functional on all arch (with exception of now `comcerto2k` but looking into it).

I might take a stab at creating a subsequent PR to create an individual `squid` package as I am used to work with it as my default defacto proxy solution.

In the meantime, pending `comcerto2k` fix, has anyone of you able to do another round of review before I merge this up?  Thnx in advance.

**Comment by defaultsecurity** [comment]
Any prospects regarding ruTorrent?

**Comment by publicarray** [comment]
> Any prospects regarding ruTorrent?

Not that I'm aware of. The pr description is pretty clear on that:

> This PR is not intented to make the package to actually work on neither DSM6/7 but solely focus on removing build errors.

I would consider better alternatives like download station or transmission

**Comment by th0ma7** [comment]
- `deluge` was ported to DSM7
- `rutorrent` as well, to be published shortly https://github.com/SynoCommunity/spksrc/pull/5455#issuecomment-1438876639
- As for `ffsync` I doubt it will ever be as stuck on python2 but the rust alternative may be of interest https://github.com/mozilla-services/syncstorage-rs in the future
- and `squidguard`... need someone interested to look into it.

**Pull request merged**